src/reaktoro_pse/core/util_classes/rkt_inputs.py

```diff
@@ -65,19 +90,32 @@ def __init__(self, var_name, pyomo_var=None):
         self.required_unit = None
         self.rkt_name = var_name
         self.lower_bound = None
+        self.upper_bound = None
         self.input_type = None
+        self.io_type = None  # input or output
+        self.dummy_var = None
+        self.dummy_var_key = None
         if pyomo_var is not None:
-            if isinstance(pyomo_var, (Var, VarData)) == False:
+            if isinstance(pyomo_var, DummyPyomoVar):
+                # if its a dummy variable, we do not need to set it
+                self.pyomo_var = None
+                self.dummy_var = pyomo_var
+                self.value = self.dummy_var.value
+                self.dummy_var_key = self.dummy_var.original_key
+            elif isinstance(pyomo_var, (Var, VarData, ScalarVar)):
```

Comment:
Checking if it's a `VarData` should be enough here. And a `Var` might be indexed, in which case you can't ask for its `value`.

Reply:
Hmm done. I wonder why I added those options. Maybe i did Var first,  and had issues with indexed vars, so I tacked on VarData and ScalarVar, because why not. I removed it to just VarData now as per suggestion.